### ISSUE TITLE
Installation on Ubuntu 18.04 won't work

### ISSUE THREAD

**Issue by woolyseal** [ploneintranet/ploneintranet#913]
Hello there,

I would love to test the ploneintranet-Product, but I just can't get it running. Everything works fine until I have to do the buildout.

Then it will give me:

`Installed /tmp/easy_install-QyAZZE/bcrypt-2.0.0/.eggs/pycparser-2.19-py2.7.egg
In file included from build/temp.linux-x86_64-2.7/_bcrypt.c:493:0:
src/crypt_blowfish-1.3/ow-crypt.h:27:14: error: conflicting types for ‘crypt_r’
 extern char *crypt_r(__const char *key, __const char *setting, void *data);
In file included from src/crypt_blowfish-1.3/crypt.h:17:0,
                 from /usr/include/python2.7/Python.h:47,
                 from build/temp.linux-x86_64-2.7/_bcrypt.c:28:
/usr/include/gnu-crypt.h:60:14: note: previous declaration of ‘crypt_r’ was here
 extern char *crypt_r (const char *__key, const char *__salt,
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
An error occurred when trying to install /tmp/tmplMTFn9get_dist/bcrypt-2.0.0.tar.gz. Look above this message for any errors that were output by easy_install.`

I guess this is related to Issues like those two:
https://github.com/NixOS/nixpkgs/issues/49724
https://bitbucket.org/dholth/cryptacular/issues/11/not-installing-on-ubuntu-1804

But I don't know how to fix it from the descriptions there. Any help?

**Comment by ale-rt** [comment]
Do you have all the dependencies listed in https://github.com/ploneintranet/ploneintranet-deploy?

**Comment by woolyseal** [comment]
Yes. After not succeeding with the installation I tested the dependencies first. It seems for me that the buildout is trying to install an egg, but gets the traceback instead.

**Comment by ale-rt** [comment]
That is definitely a compilation problem.
and I can replicate it with `./bin/pip install bcrypt==2.0.0`.
But `./bin/pip install bcrypt==3.0.0` and the latest ploneintranet version (still private) uses bcrypt==3.1.4.
You might want to change the bcrypt pin in your buildout.

**Comment by woolyseal** [comment]
Okay, I guess the pinning could work, but not with the current configuration of the package. The problem I ran into next was, that dexterity.membrane needs a bcrypt Version 2 it seems. So at this point I will give up. 

I followed this documentation: https://docs.ploneintranet.org/index.html

Can't get it to work because of the pinning in versions. Just as little documentation so you know about it!

**Comment by ale-rt** [comment]
Thanks for reporting. Unluckily the public ploneintranet version is rather old.
I hope a new public release will be done soon.

**Comment by thiagonson** [comment]
I'm in the same situation. Could you solve the problem?

root@plone:~/ploneintranet-deploy# make test-docsplit
Docsplit seems to be installed OK, no errors.

The error:
Installed /tmp/easy_install-RuRRc3/bcrypt-2.0.0/.eggs/pycparser-2.19-py2.7.egg
In file included from /usr/include/python2.7/Python.h:47:0,
                 from build/temp.linux-x86_64-2.7/_bcrypt.c:28:
src/crypt_blowfish-1.3/crypt.h:17:10: fatal error: gnu-crypt.h: No such file or directory
 #include <gnu-crypt.h>
compilation terminated.
error: Setup script exited with error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
An error occurred when trying to install /tmp/tmpxej7ksget_dist/bcrypt-2.0.0.tar.gz. Look above this message for any errors that were output by easy_install.
While:
  Installing instance.
  Getting distribution for 'bcrypt==2.0.0'.

**Comment by surajprasad** [comment]
@ale-rt any update on when the new public version will be released?
If there is any update on a fix for this for ubuntu 18, kindly share. Cant wait to put this to task..
Thanks

**Comment by pilz** [comment]
Hi @surajprasad, ale-rt is on holidays, so I quickly jump in. We are currently focussing on some customer projects in dear need of digitalisation due to the corona crisis, so our work to package the next version has been slightly delayed. My best guess is that we will manage that during the summer holidays with an ETA in September, but all plans I made in the past months have gotten trashed within weeks, so this is more a hope than a promise. But we have not forgotten nor are we ignoring your need! Sorry for the delay.

**Comment by surajprasad** [comment]
@pilz Thanks for the quick update. I was going to use a ubuntu 16 on AWS to dry run this after failing on 18, but it seems like the AWS have got rid of it for all new instances. Will await this fix and release. Keep up the Great work.
Best.

**Comment by pilz** [comment]
Thank you! Will do. And the new one will work on a recent ubuntu of course :)

**Comment by surajprasad** [comment]
@pilz possible to request for demos during this time, or a standalone way this could be tried out on my local MacBook? I intend to use Quaive in tandem with a block chain algo, which I am currently working on, wanted to take a look at how things are structured internally and if I could use perhaps existing hooks to help build the algo. Let me know. 
Best.

**Comment by pilz** [comment]
Yes, of course! I can create an account for you on a shared demo server, would that help? Just out of curiosity, how do you plan to couple an intranet with blockchain? What is the goal?

**Comment by surajprasad** [comment]
Still planning on that, in conceptual stages. Would love to bounce some ideas over email. Goal is to perhaps generate a reward system for performing tasks, on a separate network.

**Comment by surajprasad** [comment]
Yes, please do setup a demo account. Would you need additional details for that?
